### PR TITLE
v7.0.0 Protein-Protein Physical Interactions

### PR DESCRIPTION
**Pull request by Onariaginosa**
New Feature:
    * Protein Protein Physical Interactions Database and UI (#1028, #1020)
Enhancements:
    * Change order of Yeastmine versions in generate GRN modal (#1035)
    * Filename and display nomenclature for YeastMine-generated Networks (#983)
Documentation:
    * Database setup in Initial Setup wiki (#1011)

### CODE REVIEW

File: server/dals/network-dal.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ const convertResponseToJSON = function (queryType, query, totalOutput) {
             const timestamp = x.time_stamp;
             const source = x.source;
             const displayName = x.display_name;
-            JSONOutput.sources[`${displayName} : ${timestamp.toISOString().split("T")[0]}`] = {timestamp, source};
+            JSONOutput.sources[`${displayName}: ${timestamp.toISOString().split("T")[0]}`] = {timestamp, source};
```
Comment (dondi) [inline server/dals/network-dal.js:69]: Nice attention to detail 😊

File: web-client/public/js/generateNetwork.js [hunk 2]
```diff
@@ -94,34 +104,58 @@ export const generateNetwork = function () {
...
+                        displayCurrentGenes();
+                    } else {
+                        alert(
+                            `${userGeneProtein}: "${searchGeneProtein}" was not found in this database. Please check for any typos and try again.`
+                        );
+                    }
+                }).catch(function (error) {
+                    console.log(error.stack);
+                    console.log(error.name);
+                    console.log(error.message);
```
Comment (dondi) [inline web-client/public/js/generateNetwork.js:156]: Mental note for future work: we should probably place this information in our standard error/warning modal right?

File: web-client/public/js/generateNetwork.js [hunk 3]
```diff
@@ -143,15 +177,26 @@ containing "-", "_", and alpha-numeric characters only`);
...
+                geneRegulation: null
+            }
         };
     // get sources from database
+        queryProteinProteinDatabase({type:"NetworkSource"}).then(function (response) {
+            grnState.customWorkbook.sources.proteinProteinInteractions = response.sources;
+        }).catch(function (error) {
+            console.log(error.stack);
+            console.log(error.name);
+            console.log(error.message);
```
Comment (dondi) [inline web-client/public/js/generateNetwork.js:193]: More of the same…

File: web-client/public/js/update-app.js [hunk 4]
```diff
@@ -477,24 +481,21 @@ const expandLayoutSidebar = () => {
...
-//     }
-// };
+const toggleLayout = (on, off) => {
+    if (!$(on).prop("checked")) {
+        $(on).prop("checked", true);
+        $(off).prop("checked", false);
+        $(`${off} span`).removeClass("glyphicon-ok");
+        $(`${on} span`).addClass("glyphicon-ok");
+    }
+};
```
Comment (dondi) [inline web-client/public/js/update-app.js:491]: Wait—was the intent here to delete the code outright? I seem to recall this was part of @ntran18’s pull request…

**Review by dondi — APPROVED** [review]
Generally LGTM, with a couple of non-blocking comments. The error reporting can be done for another day, but please check if @ntran18’s work is integrated correctly here (mainly whether `toggleLayout` legit needs to be restored or should actually be removed)

Thanks—great work!

File: web-client/public/js/update-app.js (hunk 4 above)
Comment (Onariaginosa) [reply web-client/public/js/update-app.js:491]: Nope! So that function toggles between whether two items are checked in the top bar menu. While @ntran18 removed it because the Force Graph and Grid Layout were considered to be buttons that provoked an action rather than part of the graphs state. However, when switching between protein-protein mode and grn mode, this is a state change and the mode should be indicated by the check mark. So I brought back this function and used it for the network type switcher. 

In short it was deleted because it wasn't used for the bug, but I ended up using the pre-existing function in my new code

Comment (dondi) [reply web-client/public/js/update-app.js:491]: Got it, thanks for clarifying that!
